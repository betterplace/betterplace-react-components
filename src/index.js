export * from './components/project_teaser'
export * from './components/share_buttons'
export * from './helpers/color_helper'
export * from './helpers/format_amount'
export * from './helpers/sharing_helper'
export * from './preview'
