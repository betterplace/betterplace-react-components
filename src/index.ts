export { ProjectTeaser } from './components/project_teaser'
export { FundraisingEventTeaser } from './components/fundraising_event_teaser'

// sharing
export { emailShareAction, EmailButton } from './components/share_buttons/email_button'
export { facebookShareAction, FacebookButton } from './components/share_buttons/facebook_button'
export {
  facebookMessengerShareAction,
  FacebookMessengerButton,
} from './components/share_buttons/facebook_messenger_button'
export { telegramShareAction, TelegramButton } from './components/share_buttons/telegram_button'
export { twitterShareAction, TwitterButton } from './components/share_buttons/twitter_button'
export { whatsappShareAction, WhatsappButton } from './components/share_buttons/whatsapp_button'
export { linkAction, LinkButton } from './components/share_buttons/link_button'
export { instagramAction, InstagramButton } from './components/share_buttons/instagram_button'

// helpers
export { changeBrightness } from './helpers/color_helper'
export { formatAmount } from './helpers/format_amount'
export { encodedShareURL, openPopup } from './helpers/sharing_helper'
