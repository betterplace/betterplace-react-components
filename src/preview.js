import React from 'react'
import { render } from 'react-dom'
import { ProjectTeaser } from './components/project_teaser'

const teaserNode = document.getElementById('brc-preview-teaser')

teaserNode &&
fetch('https://www.bp42.com/api_v4/projects/1114.json')
  .then(response => response.json())
  .then(project =>
    render(
      <ProjectTeaser
        project={project}
        showDescription={true}
        progressbarColor='#9ECB0A'
        textColor='#000000'
      />,
      teaserNode
    )
  )
