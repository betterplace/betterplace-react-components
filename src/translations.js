import I18n from 'i18n-js'
global.I18n = global.I18n || I18n

I18n.locale = document.documentElement.lang || 'de'
I18n.translations = I18n.translations || {}

// pass existing translations as second arg so that they are not overridden
I18n.translations = I18n.extend({
  de: {
    bp_project_teaser: {
      values_donor_count: "Spenden",
      values_open_amount_in_cents: "fehlen noch"
    }
  },
  en: {
    bp_project_teaser: {
      values_donor_count: "donations",
      values_open_amount_in_cents: "still needed"
    }
  }
}, I18n.translations)
