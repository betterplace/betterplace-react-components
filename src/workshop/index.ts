export * from './timer_widget';
