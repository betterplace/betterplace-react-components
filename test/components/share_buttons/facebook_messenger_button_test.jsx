import { FacebookMessengerButton } from '../../../src'

const shareUrl = 'https://foo.bar'
const shape = 'round'
const facebookAppId = 'markzuckerberg'

describe('FacebookMessengerButton', () => {
  it('renders a round button', () => {
    const element = shallow(<FacebookMessengerButton shareUrl={shareUrl} shape={shape} facebookAppId={facebookAppId} />)
    expect(element).toMatchSnapshot()
  })

  it('does render if boxShadow is true', () => {
    const element = shallow(<FacebookMessengerButton shareUrl={shareUrl} shape={shape} facebookAppId={facebookAppId} boxShadow/>)
    expect(element).toMatchSnapshot()
  })

  it('does render if utmParams are provided', () => {
    const element = shallow(<FacebookMessengerButton shareUrl={shareUrl} shape={shape} facebookAppId={facebookAppId} utmParams={{ utm_campaign: 'user_share', utm_medium: 'foo', utm_source: 'bar' }}/>)
    expect(element).toMatchSnapshot()
  })

  it('does render if teaser are provided', () => {
    const element = shallow(<FacebookMessengerButton shareUrl={shareUrl} shape={shape} facebookAppId={facebookAppId} teaser='Hello there'/>)
    expect(element).toMatchSnapshot()
  })

  it('adds a beforeClick function for the click handler', () => {
    const element = shallow(<FacebookMessengerButton shareUrl={shareUrl} shape={shape} facebookAppId={facebookAppId} beforeOnClick={() => 'hey there'}/>)
    expect(element).toMatchSnapshot()
  })
})
