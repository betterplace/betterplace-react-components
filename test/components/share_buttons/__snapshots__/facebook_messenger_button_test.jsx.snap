// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookMessengerButton adds a beforeClick function for the click handler 1`] = `
<styled.a
  activeColor="#005cd7"
  aria-label="Facebook Messenger"
  color="#0084FF"
  hoverColor="#0070eb"
  onClick={[Function]}
  role="button"
  title=""
>
  <FacebookMessengerIcon />
</styled.a>
`;

exports[`FacebookMessengerButton does render if boxShadow is true 1`] = `
<styled.a
  activeColor="#005cd7"
  aria-label="Facebook Messenger"
  boxShadow={true}
  color="#0084FF"
  hoverColor="#0070eb"
  onClick={[Function]}
  role="button"
  title=""
>
  <FacebookMessengerIcon />
</styled.a>
`;

exports[`FacebookMessengerButton does render if teaser are provided 1`] = `
<styled.a
  activeColor="#005cd7"
  aria-label="Facebook Messenger"
  color="#0084FF"
  hoverColor="#0070eb"
  onClick={[Function]}
  role="button"
  title=""
>
  <FacebookMessengerIcon />
</styled.a>
`;

exports[`FacebookMessengerButton does render if utmParams are provided 1`] = `
<styled.a
  activeColor="#005cd7"
  aria-label="Facebook Messenger"
  color="#0084FF"
  hoverColor="#0070eb"
  onClick={[Function]}
  role="button"
  title=""
>
  <FacebookMessengerIcon />
</styled.a>
`;

exports[`FacebookMessengerButton renders a round button 1`] = `
<styled.a
  activeColor="#005cd7"
  aria-label="Facebook Messenger"
  color="#0084FF"
  hoverColor="#0070eb"
  onClick={[Function]}
  role="button"
  title=""
>
  <FacebookMessengerIcon />
</styled.a>
`;
